### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by danitaanubhuti**
**Pull Request: Classifier for Disease Classification**
Report - Pattern Recognition (COMP 3710 - semester 2, 2023)
**Summary:**

This pull request introduces a classifier based on a Siamese network  for the purpose of disease classification. The goal is to classify Alzheimer's disease within the ADNI brain dataset . For this purpose,  I have implemented a CNN classifier that utilizes a Siamese network structure. This classifier is designed to classify the presence or absence of Alzheimer's disease within the ADNI brain dataset. 

**Additional Details:**

Path to the accessing all the python codes and readme : recognition/project_s4745511 
More information on the working of the codes and  about the dataset is written in the Readme

Name: Danita Anubhuti Prakash
student ID: s4745511

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: Very good
- clear flow of information
- Algortihm and training details explained well
- Relevant plots for loss, accuracy provided
- results followed with discussion

Commit messages: Good

Code: 
- design is ok
- well commented
- algorithm seem to be correctly implemented

Functionality/Performance: 
- siamese seems to be working OK
- classifier is massively unstablefor the validation set, although no action have been taken to resolve this
- classifier accuracy on test set is 66%

General comments:
- algorithm solves the problem apporpriately although performance seems to be poor

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
